app/src/main/java/fc/be/app/domain/place/service/PlaceService.java

```diff
@@ -91,9 +95,20 @@ public PlaceNearbyResponse bringNearbyPlaces(PlaceNearbyRequest placeNearbyReque
         return PlaceNearbyResponse.from(places).with(apiRatingResponses);
     }
 
-    public PlacePopularGetResponse bringPopularPlaces(int numOfRows) {
-        var popularPlaces = popularPlacesResponse.places();
-        return new PlacePopularGetResponse(popularPlaces.subList(0, Math.min(numOfRows, popularPlaces.size())));
+    public PlacePopularGetResponse bringPopularPlaces(PlacePopularGetRequest placePopularGetRequest) {
+        Supplier<Stream<PlaceDTO>> streamSupplier = () -> popularPlaces.stream();
+
+        Stream<PlaceDTO> filteredStream = streamSupplier.get();
+
+        if (placePopularGetRequest.placeTypeId() != null) {
+            filteredStream = filteredStream.filter(placeDTO ->
+                    Objects.equals(placeDTO.getContentTypeId(), placePopularGetRequest.placeTypeId())
+            );
+        }
+
+        List<PlaceDTO> places = filteredStream.limit(placePopularGetRequest.size()).toList();
+
+        return PlacePopularGetResponse.from(places);
```

Comment:
분기처리하신 의도는 타당해보입니다. filter라는 불필요한 파이프라인이 추가된다면 오버헤드에 대한 걱정이 생길 수 밖에 없죠.
그런데 생각보다 파이프라인에 대한 오버헤드가 적습니다! 차라리 이런식으로 가독성을 높이는게 더 좋아보일 수 있을 것 같네요 😄 
```suggestion
               List<PlaceDTO> places = popularPlaces
                .stream()
                .filter(placeDTO -> placePopularGetRequest.placeTypeId() != null
                        && Objects.equals(placeDTO.getContentTypeId(), placePopularGetRequest.placeTypeId()))
                .limit(placePopularGetRequest.size())
                .toList();
        return PlacePopularGetResponse.from(places);
```

Reply:
@Nine-JH 반영했습니다. 감사합니다 🙇